src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -277,6 +286,39 @@ private IList<ContentItem> FindItemsImplementation(PatternSet definition, IEnume
             return itemsList;
         }
 
+        private string GetRelatedFileExtensionProperty(ContentItem contentItem, IEnumerable<Asset> assets)
+        {
+            //E.g. if path is "lib/net472/A.B.C.dll", the prefix will be "lib/net472/A.B.C."
+            string assemblyPrefix = contentItem.Path.Substring(0, contentItem.Path.LastIndexOf('.') + 1);
+
+            List<string> relatedFileExtensionList = null;
+            foreach (var asset in assets)
+            {
+                if (asset.Path is not null &&
+                    asset.Path != contentItem.Path &&
+                    asset.Path.StartsWith(assemblyPrefix, StringComparison.Ordinal))
+                {
+                    if (relatedFileExtensionList is null)
+                    {
+                        relatedFileExtensionList = new List<string>();
+                    }
+                    relatedFileExtensionList.Add(asset.Path.Substring(assemblyPrefix.Length - 1));
+                }
+            }
+
+            // If no related files found.
+            if (relatedFileExtensionList is null || !relatedFileExtensionList.Any())
+            {
+                return null;
+            }
+            else
+            {
+                relatedFileExtensionList.Sort();
+                string relatedFileExtensionsProperty = String.Join(";", relatedFileExtensionList.ToArray());
```

Comment:
```suggestion
                string relatedFileExtensionsProperty = String.Join(";", relatedFileExtensionList);
```

[`string.Join` has an overload that accepts `IEnumerable<string>`](https://docs.microsoft.com/en-us/dotnet/api/system.string.join?view=net-6.0#system-string-join(system-string-system-collections-generic-ienumerable((system-string)))).

`.ToArray()` causes many memory allocations, and `.ToList()` is usually preferable when needed. But in this case, we don't need to convert to any other data structure.